src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassUpdateServiceImpl.java

```diff
@@ -277,7 +277,13 @@ protected boolean shouldKeepWaitingForDocument(String documentId) {
         Class<R> businessObjectClass = objectDefinition.getBusinessObjectClass();
         R businessObject = businessObjectSupplier.get();
         List<String> missingRequiredFields = new ArrayList<>();
+        boolean isNewObject = StringUtils.equalsIgnoreCase(maintenanceAction, KRADConstants.MAINTENANCE_NEW_ACTION);
+        boolean isObjectEdit = StringUtils.equalsIgnoreCase(maintenanceAction, KRADConstants.MAINTENANCE_EDIT_ACTION);
+        
         for (RassPropertyDefinition propertyMapping : objectDefinition.getPropertyMappings()) {
+            if (isObjectEdit && propertyMapping.isSkipForObjectEdit()) {
+                continue;
+            }
```

Comment:
I had set up the condition in this manner so that I wouldn't have to use an "else" block to indent the rest of the loop logic. I could change this to put the remaining logic in an indented block, but If I do that then it would probably make more sense to just make this an "if" block without an "else" (and invert the condition accordingly). Please let me know if you think such a change would make this more readable.

Reply:
After thinking through this further, I'll work on restructuring the code so that the edit-mode-based filtering will occur elsewhere, and so that it doesn't need to rely on the "continue" command.